lib/LANraragi/Plugin/Metadata/EHentai.pm

```diff
@@ -101,28 +102,44 @@ sub get_tags {
         $logger->debug("EH API Tokens are $gID / $gToken");
     }
 
-    my ( $ehtags, $ehtitle ) = &get_tags_from_EH( $lrr_info->{user_agent}, $gID, $gToken, $jpntitle, $additionaltags );
-    my %hashdata = ( tags => $ehtags );
+    my $json = &get_tags_from_EH( $lrr_info->{user_agent}, $gID, $gToken );
+    my %hashdata = ( tags => "" );
+
+    if ( $json ) {
+        my $tags = &get_tags_from_json( $json, $additionaltags );
+        # only check language on e-hentai
+        if ( !$enablepanda and !has_language_tag($tags) ) {
+            my $dom = &get_EH_gallery_dom($lrr_info->{user_agent}, $domain, $gID, $gToken);
+            if ($dom) {
+                my $language = &get_language_from_dom($dom) ;
+                if ($language) { push( @$tags, "language:$language"  ); }
+            }
+        }
 
-    # Add source URL and title if possible/applicable
-    if ( $hashdata{tags} ne "" ) {
+        if ( !$hasSrc ) { push( @$tags, "source:" . get_source_url( $domain, $gID, $gToken ) ); }
+        if ($savetitle) { $hashdata{title} = get_title_from_json( $json, $jpntitle ); }
 
-        if ( !$hasSrc ) { $hashdata{tags} .= ", source:" . ( split( '://', $domain ) )[1] . "/g/$gID/$gToken"; }
-        if ($savetitle) { $hashdata{title} = $ehtitle; }
+        $hashdata{tags} = join(', ', @$tags);
+        $logger->info("Sending the following tags to LRR: " . $hashdata{tags});
     }
 
     #Return a hash containing the new metadata - it will be integrated in LRR.
     return %hashdata;
 }
 
+sub get_local_logger {
+    my %pi = plugin_info();
+    return get_logger( $pi{name}, "plugins" );
```

Comment:
I'm too lazy to do copy&paste :)

Anyway I think it's preferable to create a "`Const::Fast`" hash to pass with `plugin_info()`, so it's easier to use inside the module.

But as I said, I tried to contain myself because I don't know what's behind the plugins.

Reply:
I was thinking of eventually making a generic Util method that plugins could call to get a logger by passing their own instance (so we could dynamically call `plugin_info` and get the name there), but the const method could also work! Although it'd bring an extra module dependency. 😁